zipline/utils/run_algo.py

```diff
@@ -148,7 +148,13 @@ def _run(handle_data,
                 "invalid url %r, must begin with 'sqlite:///'" %
                 str(bundle_data.asset_finder.engine.url),
             )
-        env = TradingEnvironment(asset_db_path=connstr, environ=environ)
+        env = TradingEnvironment(
```

Comment:
Why the changes here? (I'm about to remove this class entirely, so I'm surprised to see nontrivial changes involving it.)

Reply:
This was also for test_examples. Previously we ignore the cached benchmark data if it doesn't cover the requested date range, but that date range was always the entire trading calendar (`trading_calendar.schedule.index`). Now it's filtered to the range of the backtest, so we can use the cached benchmarks.